hubblestack/extmods/modules/hstatus.py

```diff
@@ -11,12 +16,50 @@ def get():
 
         (probably only useful from other excution modules)
     '''
-    return hubblestack.status.HubbleStatus.as_json()
+    return hubblestack.status.HubbleStatus.stats()
+
+def msg_counts(pat=r'hubblestack.hec.obj.input:(?P<stype>[^:]+)', reset=True):
+    ''' returns counter data formatted for the splunk_generic_return returner
+
+        params:
+            pat - the key matching algorithm is a simple regular expression
+    '''
+
+    km = re.compile(pat)
+    r = list()
+    s = get()
+    fudge_me = None
+    rst = set()
+    for k,v in s.iteritems():
+        try:
+            if v['first_t'] == 0 or v['last_t'] == 0:
+                continue
+        except KeyError:
+            continue
+        m = km.match(k)
+        if m:
+            d = m.groupdict()
+            d.update({ 'count': v['count'], 'start': int(v['first_t']),
+                'end': int(math.ceil(v['last_t'])) })
+            r.append(d)
+            if d['stype'] == 'hubble_hec_counters':
+                fudge_me = d
+        rst.add(k)
+
+    if r:
+       #for k in rst:
+       #    hubblestack.status.HubbleStatus.reset(k)
+        min_time = min([ x['start'] for x in r ])
+        if fudge_me:
+            # this is a fudge factor for the recursion where we report on own
+            # sourcetype missing the currently-being-submitted counts
+            fudge_me['count'] += len(r)
+        return { 'sourcetype': 'hubble_hec_counters', 'time': min_time, 'events': r }
 
 def dump():
     ''' trigger a dump to the status.json file as described in hubblestack.status
 
         This is intended to be invoked from a daemon schedule and is probably
         not useful outside that context.
     '''
-    return hubblestack.status.HubbleStatus.dumpster_fire()
+    hubblestack.status.HubbleStatus.dumpster_fire()
```

Comment:
I'm thinking we should return this data. At least that way it will show up in the debug logs -- just throwing it away is probably not the best move.